### DISCUSSION TITLE
M3 screw in adapter for xiao , best for 3d printed non PCB projects

### DISCUSSION THREAD

**Discussion by gokuxmaker** [Seeed-Studio/OSHW-XIAO-Series#12] [Wish List]
![image](https://github.com/Seeed-Studio/XIAO_Series/assets/64944561/c99fc4d1-0b1e-43b6-86a6-7d85c0d83e0f)

I had trouble fitting the Xiao into my 3D printed projects because it doesn't have any screw holes. While glue is an option for some projects, it's not ideal if we need to regularly connect and disconnect the USB. Some makers don't require a custom PCB for simple projects, so the only option for them to securely fit the Xiao in the project is to glue it. Additionally, if we need to connect the battery in the Xiao ESP32C3, we have to solder wires on the back side of the Xiao, which prevents it from laying flat on a 3D printed surface. 
just in order to solve this problem I come up with this idea , I small PCB adapter . 
![Screenshot 2024-06-22 084837](https://github.com/Seeed-Studio/XIAO_Series/assets/64944561/8b940e02-0efd-4365-95c8-e37465126275)
![Screenshot 2024-06-22 093659](https://github.com/Seeed-Studio/XIAO_Series/assets/64944561/d0949f56-f342-4d75-9ff6-a921df950305)
![Screenshot 2024-06-22 084907](https://github.com/Seeed-Studio/XIAO_Series/assets/64944561/f1e94981-090e-446f-9bab-fb0f396a097e)
![Screenshot 2024-06-22 093710](https://github.com/Seeed-Studio/XIAO_Series/assets/64944561/9bca67df-2820-408b-8bbf-2fb58bd6829d)
A compact PCB which can be used to solder the xiao. There should be two 3mm holes on both sides for M3 screws. Additionally, there should be a small hole on the back side for connecting the battery contacts to the PCB. The battery terminal has been connected to soldering pads on the front bottom of the PCB, making it easy for the user to solder the battery wires.
**Why M3 screws ?**
 I suggest using M3 screws for this PCB due to the wide availability of the screws. If we use M2 screws, the chance of finding them at a local shop is low. M2 screws are commonly used in electronics, whereas M3 screws are widely used in automotive, making them easier to find both online and at local shops (based on my personal experience).

so this is my idea let me know your thoughts

**Comment by ekuleshov** [comment]
@gokuxmaker couldn't you instead make a en enclosure that would hold Xiao board at 4 corners and then screw-in that enclosure?

**Reply by gokuxmaker** [reply]
it will be easy to assemble. Using a PCB, it allows for easy access to the underside pin contacts of Xiao. The presence of larger soldering pads will make it easier to connect wires for DIY projects, so this type of adapter can be used for that purpose as well."

**Reply by rootexpression** [reply]
I also prefer the PCB solution be an available option, because it provides more robust mounting. In addition, it'd be significantly easier to design around for most people, and lowers the bar, helping them to get their project done, and get on to the next one. Which helps boost their activity in both hobbies (3D printing, and working with micro-controllers).
That said, it wouldn't be too tough to make a step file that handles holding on to the board which can be imported into projects (though alignment can be more difficult with complex/freeflowing designs without a good reference, like when a model is based on a 3D scan).
